pkgs/racket-doc/scribblings/guide/other-editors.scrbl

```diff
@@ -137,91 +137,75 @@ popular among Racketeers as well.
-Vim detects files with the extension @tt{.rkt} as having the
-@tt{scheme} filetype. @hyperlink["https://github.com/vim/vim/commit/9cd91a1e8816d727fbdbf0b3062288e15abc5f4d"]{Version 8.2.3368}
-added support for @tt{.rktd} and @tt{.rktl}.
+for Racket. Vim also ships with some special support for Racket.
 
-In older versions, you can enable filetype detection of Racket
-files as Scheme with the following:
+The @tt{racket} filetype comes with
+@itemlist[
+  @item{syntax highlighting}
+  @item{custom indentation for Racket forms}
+  @item{and other support including comments and @tt{raco fmt}}
+]
 
-@verbatim[#:indent 2]|{
-if has("autocmd")
-  autocmd filetypedetect BufReadPost *.rkt,*.rktl,*.rktd set filetype=scheme
-endif
-}|
+There is also support for several @seclink["top" #:doc '(lib "scribblings/raco/raco.scrbl")]{raco commands}
+in the form of builtin @tt{compiler} plugins; see @tt{:help compiler} for more
+information.
 
-If your Vim supports the ftdetect system, in which case it's likely new enough
-to support Racket already, you can nevertheless put the following in
-@filepath{~/.vim/ftdetect/racket.vim}
-(@filepath{$HOME/vimfiles/ftdetect/racket.vim} on MS-Windows; see @tt{:help runtimepath}).
+For information about older Vim versions, see @secref{vim-versions}.
 
-@verbatim[#:indent 2]|{
-" :help ftdetect
-" If you want to change the filetype only if one has not been set
-autocmd BufRead,BufNewFile *.rkt,*.rktl,*.rktd setfiletype scheme
-" If you always want to set this filetype
-autocmd BufRead,BufNewFile *.rkt,*.rktl,*.rktd set filetype=scheme
-}|
+@subsection[#:tag "vim-racket"]{Enhanced Racket Support}
 
-@subsection[#:tag "vim-plugins"]{Plugins}
+The following plugin is recommended for the most up-to-date experience:
+@hyperlink["https://github.com/benknoble/vim-racket"]{benknoble/vim-racket}. It
+enables auto-detection of Racket files on top of enhanced indentation and syntax
+highlighting. Vim's default support comes from a subset of this plugin;
+installing it yourself provides additional features.
 
-Alternatively, you can use a plugin such as @itemlist[
-    @item{@hyperlink["https://github.com/wlangstroth/vim-racket"]{wlangstroth/vim-racket}}
-    @item{@hyperlink["https://github.com/benknoble/vim-racket"]{benknoble/vim-racket}}
-]@margin-note{The major difference between the two is that the
-@tt{benknoble/vim-racket} fork supports more features out of the box and is
-updated more frequently.}
-to enable auto-detection, indentation, and syntax highlighting specifically for
-Racket files.
+@subsection{Filetypes}
 
-These plugins work by setting the @tt{filetype} option based on the @(hash-lang)
+Vim will detect your Racket files as Scheme out of the box. To get the
+additional features of the Racket filetype, consider installing the plugin from
+@secref{vim-racket}.
+
+The @tt{vim-racket} plugin detects the @tt{filetype} option based on the @(hash-lang)
```

Comment:
This section now refers to the preceding section a lot - can they be combined? For example:

```
# Enhanced Racket Support

Vim will detect your Racket files as Scheme out of the box. To get the additional features of the Racket filetype, consider installing the @tt{vim-racket} plugin.

This plugin enables auto-detection of Racket files on top of enhanced indentation and syntax highlighting. Vim's default support comes from a subset of this plugin;
installing it yourself provides additional features.
...
```

... followed by the explanation of filetype selection and other content in this section.

Reply:
Diff:
```patch
diff --git a/pkgs/racket-doc/scribblings/guide/other-editors.scrbl b/pkgs/racket-doc/scribblings/guide/other-editors.scrbl
index 92f19c5979..2662c9431f 100644
--- a/pkgs/racket-doc/scribblings/guide/other-editors.scrbl
+++ b/pkgs/racket-doc/scribblings/guide/other-editors.scrbl
@@ -154,18 +154,14 @@ For information about older Vim versions, see @secref{vim-versions}.
 
 @subsection[#:tag "vim-racket"]{Enhanced Racket Support}
 
-The following plugin is recommended for the most up-to-date experience:
+Vim will detect your Racket files as Scheme out of the box. To get the
+additional features of the Racket filetype, consider installing the
+@tt{vim-racket} plugin from
 @hyperlink["https://github.com/benknoble/vim-racket"]{benknoble/vim-racket}. It
 enables auto-detection of Racket files on top of enhanced indentation and syntax
 highlighting. Vim's default support comes from a subset of this plugin;
 installing it yourself provides additional features.
 
-@subsection{Filetypes}
-
-Vim will detect your Racket files as Scheme out of the box. To get the
-additional features of the Racket filetype, consider installing the plugin from
-@secref{vim-racket}.
-
 The @tt{vim-racket} plugin detects the @tt{filetype} option based on the @(hash-lang)
 line. For example:@itemlist[
     @item{A file starting with @code{#lang racket} or @code{#lang racket/base} has @tt{filetype} equal to @tt{racket}.}
@@ -182,10 +178,8 @@ other Vim users will benefit.
 
 @subsection{Indentation}
 
-If you use
-@hyperlink["https://github.com/benknoble/vim-racket"]{benknoble/vim-racket} and
-Vim version 9 or greater, improved indentation for the @tt{racket}
-@seclink["Filetypes"]{filetype} is configured out of the box.
+If you use @secref{vim-racket} and Vim version 9 or greater, improved
+indentation for the @tt{racket} filetype is configured out of the box.
 
 Otherwise, you can manually enable indentation for Racket by setting both the
 @tt{lisp} and @tt{autoindent} options in Vim. You will want to customize the
@@ -200,8 +194,8 @@ available on the website.
 @subsection{Highlighting}
 
 Syntax highlighting for Scheme and Racket is shipped with Vim on many platforms.
-You will want to use the @tt{racket} @seclink["Filetypes"]{filetype} for the
-best syntax experience.
+You will want to use the @tt{racket} filetype for the best syntax experience;
+see @secref{vim-racket} for enhanced syntax highlighting for Racket languages.
 
 The @hyperlink["http://www.vim.org/scripts/script.php?script_id=1230"]{Rainbow
 Parenthesis} script for Vim can be useful for more visible parenthesis
@@ -255,14 +249,15 @@ place to get help from Vimmers.
 
 @subsection[#:tag "vim-versions"]{Older Versions of Vim}
 
-As of @hyperlink["https://github.com/vim/vim/commit/9b03d3e75b4274493bbe76772d7b92238791964c"]{Version 9.0.0336},
-Vim ships with runtime files from @hyperlink["https://github.com/benknoble/vim-racket"]{benknoble/vim-racket},
-but these exclude @seclink["Filetypes"]{filetype} detection for the @tt{racket}
-filetype. If you are using this version or versions newer than this you probably
-want to tweak the suggestions in this document to use the @tt{racket} filetype
-instead of @tt{scheme}. You should also consider installing the plugin yourself
-to get the latest changes, since Ben is slow to sync changes upstream to Vim and
-since the plugin contains improved filetype detection.
+As of
+@hyperlink["https://github.com/vim/vim/commit/9b03d3e75b4274493bbe76772d7b92238791964c"]{Version 9.0.0336},
+Vim ships with runtime files from @secref{vim-racket}, but these exclude
+filetype detection for the @tt{racket} filetype. If you are using this version
+or versions newer than this you probably want to tweak the suggestions in this
+document to use the @tt{racket} filetype instead of @tt{scheme}. You should also
+consider installing the plugin yourself to get the latest changes, since Ben is
+slow to sync changes upstream to Vim and since the plugin contains improved
+filetype detection.
 
 As of @hyperlink["https://github.com/vim/vim/commit/1aeaf8c0e0421f34e51ef674f0c9a182debe77ae"]{version 7.3.518},
 Vim detects files with the extension @tt{.rkt} as having the
```